test/FunctionalTests/Client/StreamingTests.cs

```diff
@@ -996,5 +997,90 @@ async Task WaitForAllStreams(IAsyncStreamReader<DataMessage> requestStream, ISer
             }
         }
 #endif
+
+        [TestCase(2)]
```

Comment:
I might've missed the nuance in the issue text but why are we testing on iterations up to `2` and `20`?

Reply:
The race condition happened randomly.

When I was testing, I had 2 iterations so the logs weren't too long in one test run, and 20 in another run to ensure the race condition didn't happen over many iterations.